addons/website/models/website_page.py

```diff
@@ -239,26 +239,21 @@ def _search_fetch(self, search_detail, search, limit, order):
         if with_description and search:
             # Perform search in translations
             # TODO Remove when domains will support xml_translate fields
-            query = sql.SQL("""
-                SELECT DISTINCT {table}.id
-                FROM {table}
-                LEFT JOIN ir_ui_view v ON {table}.view_id = v.id
-                WHERE (v.name ILIKE {search}
-                OR COALESCE(v.arch_db->>{lang}, v.arch_db->>'en_US') ILIKE {search})
-                AND {table}.id IN {ids}
-                LIMIT {limit}
-            """).format(
-                table=sql.Identifier(self._table),
-                search=sql.Placeholder('search'),
-                lang=sql.Literal(self.env.lang or 'en_US'),
-                ids=sql.Placeholder('ids'),
-                limit=sql.Placeholder('limit'),
-            )
-            self.env.cr.execute(query, {
-                'search': '%%%s%%' % escape_psql(search),
-                'ids': tuple(most_specific_pages.ids),
-                'limit': len(most_specific_pages.ids),
-            })
+            self.env.cr.execute(SQL("""
+                SELECT DISTINCT %(table)s.id
+                FROM %(table)s
+                LEFT JOIN ir_ui_view v ON %(table)s.view_id = v.id
+                WHERE (v.name ILIKE %(search)s
+                OR COALESCE(v.arch_db->>%(lang)s, v.arch_db->>'en_US') ILIKE %(search)s)
+                AND %(table)s.id IN %(ids)s
+                LIMIT %(limit)s
+                """,
+                table=SQL.identifier(self._table),
+                search='%%%s%%' % escape_psql(search),
```

Comment:
```suggestion
                search=f"%{escape_psql(search)}%",
```
I have the feeling that escaping at this level is wrong...